ansible/playbooks/roles/repository/files/download-requirements/src/command/dnf.py

```diff
@@ -1,9 +1,13 @@
 from typing import Dict, List
+import re
 
 from src.command.command import Command
 from src.error import CriticalError
 
 
+def filter_non_critical_errors(stderr: str):
+    re.sub('^Failed to set locale, defaulting to .+\n', '', stderr)
```

Comment:
This function does not return anything, It doesn't work. Strings are passed by value and also are immutable, therefore re.sub() returns new string which in this case is not returned.

I suggest returning a list of strings (it's faster).
```python
return [line for line in '\n'.join(stderr) if not line.startswith('Failed to set locale, defaulting to')]
```